uplc-syntax.md

```diff
@@ -34,12 +34,30 @@ module UPLC-CONCRETE-SYNTAX
                         | "bytestring"
                         | "unit"
                         | "bool"
+                        | "list" "(" TypeConstant ")"
+                        | "pair" "(" TypeConstant ")" "(" TypeConstant ")"
 
   syntax Constant ::= Int
                     | "True"
                     | "False"
                     | ByteString
                     | "()"
+                    | "[" ConstantList "]"
+                    | "(" Constant "," Constant ")"
+                    | "{" TextualData "}"
+
+  syntax ConstantList ::= List{Constant, ","}
+
+  syntax TextualData ::= "Constr" Int "[" DataList "]"
+                       | "Map" "[" DataPairList "]"
+                       | "List" "[" DataList "]"
+                       | "Integer" Int
+                       | "ByteString" ByteString
+
```

Comment:
This all looks sensible.  The idea of enclosing the contents of a Data object inside `{...}` should make it a lot easier to parse since it's easy to work out where it comes to an end.  Be aware that in typed Plutus Core we use `{}` to denote type instantiation, for example in `[{(builtin ifThenElse) (con integer)} (con bool True) (con integer 11) (con integer 22)]`, where `ifThenElse` is polymorphic and we have to tell the type system what the return type is.  This shouldn't cause any difficulty though since in your case the `{` will always come after `data`, so there's no possibility of ambiguity.

It might be worth using `I` and `B` instead of `Integer` and `ByteString` since that's more consistent with the internal Haskell syntax, but that's a minor detail.